astropy/modeling/fitting.py

```diff
@@ -341,6 +341,129 @@ def __call__(self, model, x, y, z=None, weights=None, rcond=None):
         return model_copy
 
 
+class FittingWithOutlierRemoval(object):
+    """
+    This class combines an outlier removal technique with a fitting procedure.
+    Basically, given a number of iterations `niter`, outliers are removed in
+    and fitting is performed for each iteration.
+
+    Attributes
+    ----------
+    fitter : An Astropy fitter
+        An instance of any Astropy fitter, i.e., LinearLSQFitter,
+        LevMarLSQFitter, SLSQPLSQFitter, SimplexLSQFitter, JointFitter.
+    outlier_func : function
+        A function for outlier removal.
+    niter : int (optional)
+        Number of iterations.
+    outlier_kwargs : dict (optional)
+        Keyword arguments for outlier_func.
+    """
+
+    def __init__(self, fitter, outlier_func, niter=3, **outlier_kwargs):
+        self.fitter = fitter
+        self.outlier_func = outlier_func
+        self.niter = niter
+        self.outlier_kwargs = outlier_kwargs
+
+    @property
+    def fitter(self):
+        return self._fitter
```

Comment:
Here you return `self._fitter` but the attribute above is `self.fitter`. I think the property is ignored if this works. If you want to use `self.fitter` simply remove the `fitter` property. Similarly for the other attributes.